src/dialog.jsx

```diff
@@ -145,13 +149,19 @@ let Dialog = React.createClass({
   componentWillReceiveProps (nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+
+    if( nextProps.isOpen !== this.props.isOpen ) {
+      if( nextProps.isOpen && !this.state.open )
+        this._show();
+      else if( !nextProps.isOpen && this.state.open )
+        this._dismiss();
+    }
   },
 
   componentDidMount() {
     this._positionDialog();
     if (this.props.openImmediately) {
       this.refs.dialogOverlay.preventScrolling();
```

Comment:
I don't think that it would be called under any other conditions as was possible before.
`_show()` is only triggered if there is a real state change from closed to open. It would not be triggered when going from open to open due to some other prop changes.